synapse/storage/databases/main/e2e_room_keys.py

```diff
@@ -274,32 +274,41 @@ async def get_e2e_room_keys(
             if session_id:
                 keyvalues["session_id"] = session_id
 
-        rows = await self.db_pool.simple_select_list(
-            table="e2e_room_keys",
-            keyvalues=keyvalues,
-            retcols=(
-                "user_id",
-                "room_id",
-                "session_id",
-                "first_message_index",
-                "forwarded_count",
-                "is_verified",
-                "session_data",
+        rows = cast(
+            List[Tuple[str, str, int, int, int, str]],
```

Comment:
`first_message_index` and `forwarded_count` and `is_verified` is nullable. (The last doesn't matter we cast to a `bool` below.)

Reply:
See #16548.